src/wfs/src/main/java/org/geoserver/wfs/StoredQueryProvider.java

```diff
@@ -109,12 +152,12 @@ public StoredQuery createStoredQuery(StoredQueryDescriptionType query, boolean s
      * @param query The stored query
      */
     public void removeStoredQuery(StoredQuery query) {
-        storedQueryDir().get(toFilename(query.getName())).delete();
+        storedQueryDirByContext().get(toFilename(query.getName())).delete();
```

Comment:
What if one can see both local and global stored queries, is it not possible to remove a global one? May make sense mind, but it would be useful to get some suitable error message allowing to understand what's going on, e.g. "Cannot remove global stored queries from a local services"... however, mind, there is still the annoyance that virtual service does not tell us what's local and what's global. Seems like there is a downside no matter what path is chosen.

Reply:
Agreed, adding an exception message.